pallets/connectors/src/message.rs

```diff
@@ -56,13 +60,72 @@ where
 		address: Address,
 		valid_until: Moment,
 	},
+	// Bidirectional: Domain must not accept every incoming token.
+	// Sender must ensure beforehand that the receiver will not reject
 	Transfer {
+		token: u128,
+		source_address: Address,
+		destination_address: Address,
+		amount: Balance,
+	},
+	TransferTrancheTokens {
+		domain: Domain,
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		source_address: Address,
+		destination_address: Address,
+		amount: Balance,
+	},
+	IncreaseInvestOrder {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		address: Address,
+		token: u128,
+		amount: Balance,
+	},
+	DecreaseInvestOrder {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		address: Address,
+		token: u128,
+		amount: Balance,
+	},
+	IncreaseRedeemOrder {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		address: Address,
+		token: u128,
+		amount: Balance,
+	},
+	DecreaseRedeemOrder {
 		pool_id: PoolId,
 		tranche_id: TrancheId,
-		domain: Domain,
 		address: Address,
+		token: u128,
 		amount: Balance,
 	},
+	CollectRedem {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		address: Address,
+	},
+	CollectForRedeem {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		call_address: Address,
+		collect_address: Address,
+	},
+	CollectInvest {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		address: Address,
+	},
+	CollectForInvest {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		call_address: Address,
+		collect_address: Address,
```

Comment:
FYI, on the Solidity I am naming these `caller` and `user`, respectively. Same for the other messages. Happy to change but it would be good to have both ends aligned